subprojects/base-services/src/main/java/org/gradle/util/GUtil.java

```diff
@@ -236,11 +235,7 @@ public static Properties loadProperties(InputStream inputStream) {
     public static void saveProperties(Properties properties, File propertyFile) {
         try {
             FileOutputStream propertiesFileOutputStream = new FileOutputStream(propertyFile);
-            try {
-                properties.store(propertiesFileOutputStream, null);
-            } finally {
-                propertiesFileOutputStream.close();
-            }
+            saveProperties(properties, propertiesFileOutputStream);
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
```

Comment:
You can remove the `try-catch` here, too.

Reply:
I can't since `new FileOutputStream` may cause an `IOException`...

Reply:
Ah, right, sorry.